hashicorp/consul
# feature - [NET - 4005]  - [Supportability] Reloadable Configuration - enable_debug

### Description

Making enable_debug a Reloadable Configuration

Summary of customer problem and expectations:
In order to capture pprof to analyze Golang heap, CPU, goroutine, and trace profiling "enable_debug" must be set in the configuration and for it to take effect, Consul process must be restarted. Thereafter the "consul debug" command with the appropriate flags can be run to capture the pprof. Having the ability to apply this update to the configuration through "consul reload" would ensure cluster stability.

What currently happens or is possible:
Currently the Consul process must be restarted for this configuration to take effect.

In this PR - I am making enable_debug a reloadable configuration and adding http handlers before the enableDebug flag check. Earlier we use to check for enable debug and then add http handlers. Now I am adding http handlers and checking for enableDebug and checkACLDisabled() inside the handler when the handler is already attached.

### Testing & Reproduction steps

Replicate the issue by - 
* Git clone this repo 
* Run `make dev`
* create a config file server.json
```
{
  "node_name": "consul-server",
  "server": true,
  "bootstrap": true,
  "ui_config": {
    "enabled": true
  },
  "datacenter": "dc1",
  "data_dir": "consul/data",
  "log_level": "INFO",
  "addresses": {
    "http": "0.0.0.0"
  },
  "connect": {
    "enabled": true
  },
  "enable_debug": false
}
```

* start consul agent ` ./bin/consul agent --config-file=server.json`
* open `http://localhost:8500/v1/agent/self` -> observer `EnableDebug` is false
* change `enable_debug: true` in the server.json file
* `./bin/consul reload`
* Open `http://localhost:8500/v1/agent/self` see - EnableDebug flag does not change to true

### Testing 

* Git clone this repo 
* Run `make dev`
* create a config file server.json
```
{
  "node_name": "consul-server",
  "server": true,
  "bootstrap": true,
  "ui_config": {
    "enabled": true
  },
  "datacenter": "dc1",
  "data_dir": "consul/data",
  "log_level": "INFO",
  "addresses": {
    "http": "0.0.0.0"
  },
  "connect": {
    "enabled": false
  },
  "enable_debug": false
}
```
* start consul agent ` ./bin/consul agent --config-file=server.json`
* open `http://localhost:8500/v1/agent/self` -> observer `EnableDebug` is false
* change `enable_debug: true` in the server.json file
* `./bin/consul reload`
* Open `http://localhost:8500/v1/agent/self` see - EnableDebug flag changes to true
* Next test for PProf Run - `./bin/consul debug -capture pprof`
* Notice command runs successfully. 

### PR Checklist
